baklava/src/util.ts

```diff
@@ -1,150 +1,194 @@
+        }
+    });
+
+    ioHook.on("keydown", (event: IOHookEvent) => {
+        if (event.shiftKey) {
+            if (CURRENT_PTT_KEY.includes("Shift")) {
+                let i = CURRENT_PTT_KEY.indexOf("Shift");
+                PTT_STATUS[i] = true;
+            }
+        } else if (event.altKey) {
+            if (CURRENT_PTT_KEY.includes("Alt")) {
+                let i = CURRENT_PTT_KEY.indexOf("Alt");
+                PTT_STATUS[i] = true;
+            }
+        } else if (event.ctrlKey) {
+            if (CURRENT_PTT_KEY.includes("Control")) {
+                let i = CURRENT_PTT_KEY.indexOf("Control");
+                PTT_STATUS[i] = true;
+            }
+        } else if (event.metaKey) {
+            if (CURRENT_PTT_KEY.includes("Meta")) {
+                let i = CURRENT_PTT_KEY.indexOf("Meta");
+                PTT_STATUS[i] = true;
+            }
+        } else {
+            if (CURRENT_PTT_KEY.includes(KEY_TABLE[event.keycode - 1])) {
+                let i = CURRENT_PTT_KEY.indexOf(KEY_TABLE[event.keycode - 1]);
+                PTT_STATUS[i] = true;
+            }
+        }
+        let PTT = PTT_STATUS.every((key_status) => key_status === true);
+        if (PTT != PTT_PREV_STATUS) {
+            PTT_PREV_STATUS = PTT;
+            mainWindow.webContents.send("@voice/ptt_status_change", !PTT);
+        }
+    })
+
+    ioHook.on("keyup", (event: IOHookEvent) => {
+        if (event.shiftKey) {
+            if (CURRENT_PTT_KEY.includes("Shift")) {
+                let i = CURRENT_PTT_KEY.indexOf("Shift");
+                PTT_STATUS[i] = false;
+            }
+        } else if (event.altKey) {
+            if (CURRENT_PTT_KEY.includes("Alt")) {
+                let i = CURRENT_PTT_KEY.indexOf("Alt");
+                PTT_STATUS[i] = false;
+            }
+        } else if (event.ctrlKey) {
+            if (CURRENT_PTT_KEY.includes("Control")) {
+                let i = CURRENT_PTT_KEY.indexOf("Control");
+                PTT_STATUS[i] = false;
+            }
+        } else if (event.metaKey) {
+            if (CURRENT_PTT_KEY.includes("Meta")) {
+                let i = CURRENT_PTT_KEY.indexOf("Meta");
+                PTT_STATUS[i] = false;
+            }
+        } else {
+            if (CURRENT_PTT_KEY.includes(KEY_TABLE[event.keycode - 1])) {
+                let i = CURRENT_PTT_KEY.indexOf(KEY_TABLE[event.keycode - 1]);
+                PTT_STATUS[i] = false;
+            }
+        }
+        let PTT = PTT_STATUS.every((key_status) => key_status === true);
+        if (PTT != PTT_PREV_STATUS) {
+            PTT_PREV_STATUS = PTT;
+            mainWindow.webContents.send("@voice/ptt_status_change", !PTT);
+        }
+    });
+
+    ioHook.start();
+
+}
+
+export async function HandleVoiceTray(mainWindow: BrowserWindow) {
+    let menu = Menu.buildFromTemplate(MENU_TEMPLATE);
+    Menu.setApplicationMenu(menu);
+
+    const tray = new Tray(path.join(__dirname, `../icons/tray.png`));

```

Comment:
This should be defined (even just `let`) in the global scope. Otherwise, the garbage collector can clean up the variable and destroy the tray. 

Reference: [Electron FAQ | My app's tray disappeared after a few minutes](https://www.electronjs.org/docs/faq#my-apps-tray-disappeared-after-a-few-minutes)

Reply:
Ohk thx for the review

Reply:
also bro again. we did it again. I was also working on menu and tray and even you did XD

Reply:
np. Yeah, we should probably communicate a bit more XD

If you want to dm me on Discord, we can discuss baklava and coordinate our efforts

Reply:
@TheOtterlord sure!